core/node/external_proof_integration_api/src/processor.rs

```diff
@@ -45,17 +43,32 @@ impl Processor {
     pub(crate) async fn verify_proof(
         &self,
         Path(l1_batch_number): Path<u32>,
-        Json(payload): Json<VerifyProofRequest>,
+        mut multipart: Multipart,
     ) -> Result<(), ProcessorError> {
-        let mut guard = MethodCallGuard::new(Method::VerifyProof);
-
         let l1_batch_number = L1BatchNumber(l1_batch_number);
         tracing::info!(
             "Received request to verify proof for batch: {:?}",
             l1_batch_number
         );
 
-        let serialized_proof = bincode::serialize(&payload.0)?;
+        let mut serialized_proof = vec![];
+
+        while let Some(mut data) = multipart.next_field().await.map_err(|err| {
+            tracing::error!("Failed to read field: {:?}", err);
+            ProcessorError::InvalidProof
+        })? {
+            while let Some(chunk) = data.chunk().await.map_err(|err| {
+                tracing::error!("Failed to read chunk: {:?}", err);
+                ProcessorError::InvalidProof
+            })? {
+                serialized_proof.extend_from_slice(&chunk);
+            }
+        }
```

Comment:
Is it correct? I'm not an expert, but shouldn't we find a single exact field with the correct params (`name`/`filename`/`content-type`) and extract it only? Right now it looks like we're going through all the fields and collecting them into a single vector (which would work for forms with a single field, but is logically incorrect).

Reply:
Well, I don't think it's good idea to enforce some exact filename, but other things sound good